ldap/servers/slapd/back-ldbm/ldbm_entryrdn.c

```diff
@@ -1544,8 +1545,14 @@ _entryrdn_get_elem(entryrdn_db_ctx_t *ctx,
     /* Position cursor at the matching key */
     *elem = NULL;
     dbop = DBI_OP_MOVE_NEAR_DATA;
+
 retry_get:
+    if (*elem) {
+        /* free the previous elem data */
+        slapi_ch_free((void**)&prev_elem_data);
+    }
```

Comment:
I wonder if we also need to check and free it at the `bail:` section

Reply:
> I wonder if we also need to check and free it at the `bail:` section

Maybe :-)  Like I said this code is a bit tricky when it comes to handling long DN's vs not long DN's.  I'll try and force some failures and see what happens...